compiler/rustc_mir/src/borrow_check/invalidation.rs

```diff
@@ -92,6 +92,21 @@ impl<'cx, 'tcx> Visitor<'tcx> for InvalidationGenerator<'cx, 'tcx> {
                     self.consume_operand(location, input);
                 }
             }
+            StatementKind::CopyNonOverlapping(box rustc_middle::mir::CopyNonOverlapping {
+                ref src,
+                ref dst,
+                ref count,
+            }) => {
+                self.consume_operand(location, src);
+                self.consume_operand(location, dst);
+                self.consume_operand(location, count);
+                match dst {
+                    Operand::Move(ref place) | Operand::Copy(ref place) => {
+                        self.mutate_place(location, *place, Deep, JustWrite);
```

Comment:
I'm not sure this is correct. The `dst` is a `*mut T` so I don't believe borrock wants or needs to care about places being mutated here.

AFAIU you should just keep it as 

```rust
                self.consume_operand(location, src);
                self.consume_operand(location, dst);
                self.consume_operand(location, count); 
```

and that's it.

Reply:
**If** `CopyNonOverlapping` is only generated during intrinsic lowering, borrowck won't ever see it.

Reply:
Ah I see, if it was a `&mut T` then it would need to invalidate it?

Reply:
If it was a `&mut T` you would `mutate_place` it, yeah. Though without checking I suspect that `consume_operand` would still do the right thing for you there.